File: src/JavaJavadocComment.cs
using AutoRest.Core.Utilities;

namespace AutoRest.Java
{
    public class JavaJavadocComment
    {
        private readonly JavaFileContents contents;
        private bool expectsLineSeparator;

        public JavaJavadocComment(JavaFileContents contents)
        {
            this.contents = contents;
        }

        private void AddExpectedLineSeparator()
        {
            if (expectsLineSeparator)
            {
                expectsLineSeparator = false;
                contents.Line();
            }
        }

        public void Description(string description)
        {
            string processedText = ProcessText(description);
            if (!string.IsNullOrEmpty(processedText))
            {
                contents.Line(processedText);
                expectsLineSeparator = true;
            }
        }

        public void Param(string parameterName, string parameterDescription)
        {
            AddExpectedLineSeparator();
            contents.Line($"@param {parameterName} {ProcessText(parameterDescription)}");
        }

        public void Return(string returnValueDescription)
        {
            AddExpectedLineSeparator();
            contents.Line($"@return {ProcessText(returnValueDescription)}");
        }

        public void Throws(string exceptionTypeName, string description)
        {
            AddExpectedLineSeparator();
            contents.Line($"@throws {exceptionTypeName} {ProcessText(description)}");
        }

        private static string Trim(string value) => string.IsNullOrEmpty(value) ? value : value.Trim();

        private static string EnsurePeriod(string value) => string.IsNullOrEmpty(value) || value.EndsWith('.') ? value : value + '.';

        private static string ProcessText(string value) => EnsurePeriod(Trim(value)).EscapeXmlComment();
    }
}

File: javagen/src/main/java/com/azure/autorest/model/javamodel/JavaJavadocComment.java
package com.azure.autorest.model.javamodel;


// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.


import com.azure.autorest.util.CodeNamer;

public class JavaJavadocComment {
    private JavaFileContents contents;
    private boolean expectsLineSeparator;

    public JavaJavadocComment(JavaFileContents contents) {
        this.contents = contents;
    }

    private static String trim(String value) {
        return value == null || value.isEmpty() ? value : value.trim();
    }

    private static String ensurePeriod(String value) {
        return value == null || value.isEmpty() || value.endsWith(".") ? value : value + '.';
    }

    private static String processText(String value) {
        return CodeNamer.escapeXmlComment(ensurePeriod(trim(value)));
    }

    private void addExpectedLineSeparator() {
        if (expectsLineSeparator) {
            expectsLineSeparator = false;
            contents.line();
        }
    }

    public final void description(String description) {
        String processedText = processText(description);
        if (processedText != null && !processedText.isEmpty()) {
            contents.line(processedText);
            expectsLineSeparator = true;
        }
    }

    public final void param(String parameterName, String parameterDescription) {
        addExpectedLineSeparator();
        contents.line(String.format("@param %1$s %2$s", parameterName, processText(parameterDescription)));
    }

    public final void methodReturns(String returnValueDescription) {
        if (returnValueDescription != null && !returnValueDescription.isEmpty()) {
            addExpectedLineSeparator();
            contents.line(String.format("@return %1$s", processText(returnValueDescription)));
        }
    }

    public final void methodThrows(String exceptionTypeName, String description) {
        addExpectedLineSeparator();
        contents.line(String.format("@throws %1$s %2$s", exceptionTypeName, processText(description)));
    }
}
